Configuration/Objects.yaml

```diff
@@ -0,0 +1,24 @@
+Neos\Neos\Ui\Controller\BackendController:
+  properties:
+    configurationProvider:
+      object: Neos\Neos\Ui\Infrastructure\Configuration\ConfigurationProvider
+    routesProvider:
+      object: Neos\Neos\Ui\Infrastructure\MVC\RoutesProvider
+    frontendConfigurationProvider:
+      object: Neos\Neos\Ui\Infrastructure\Configuration\FrontendConfigurationProvider
+    nodeTypesProvider:
+      object: Neos\Neos\Ui\Infrastructure\ContentRepository\NodeTypesProvider
+    menuProvider:
+      object: Neos\Neos\Ui\Infrastructure\Neos\MenuProvider
+    initialStateProvider:
+      object: Neos\Neos\Ui\Infrastructure\Configuration\InitialStateProvider
```

Comment:
I would propose to go for a interface to className mapping instead of declaring just the properties for the backend-controller. Because the ConfigurationProvider also uses the `CacheConfigurationVersionProviderInterface` and via a global interface override it would become obvious how the dependencies are resolved.


```yaml
'Neos\Neos\Ui\Domain\NodeTypesProviderInterface':
  className: Neos\Neos\Ui\Infrastructure\ContentRepository\NodeTypesProvider
# ...
```